impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -527,6 +527,28 @@ std::unique_ptr<Surface> ContextVK::AcquireSurface(size_t current_frame) {
   return surface_producer_->AcquireSurface(current_frame);
 }
 
+#ifdef FML_OS_ANDROID
+
+vk::UniqueSurfaceKHR ContextVK::CreateAndroidSurface(
+    ANativeWindow* window) const {
+  if (!instance_) {
+    return vk::UniqueSurfaceKHR{VK_NULL_HANDLE};
+  }
+
+  auto create_info = vk::AndroidSurfaceCreateInfoKHR().setWindow(window);
```

Comment:
Don't you need to hold a reference to the ANativeWindow? Something like `ANativeWindow_acquire` and `_release`

Reply:
In my planned usage: [AndroidNativeWindow](https://github.com/flutter/engine/blob/main/shell/platform/android/surface/android_native_window.h) is going to be held by [AndroidSurfaceVulkanImpeller](https://github.com/flutter/engine/pull/37211/files#diff-86801a940180f257c85ef974265f568f40b3aee045bc4de82efb1cd8b9b75e36R48) <- is a WIP PR.